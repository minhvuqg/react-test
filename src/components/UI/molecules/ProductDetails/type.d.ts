export interface IProps {
  product: Product.Infor
}
