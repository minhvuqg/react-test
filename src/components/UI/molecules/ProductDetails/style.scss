.product-info {
  display: flex;
  width: 70%;

  .image {
    flex: 1;
    img {
      width: 100%;
      height: 100%;
      object-fit: contain;
    }
  }

  .info {
    flex: 1;
  }

  .name-and-price {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
}
