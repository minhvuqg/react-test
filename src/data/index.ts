export const productList: Product.Infor[] = [
  {
    id: '1',
    image: '/imgs/products/iphone.png',
    name: 'Iphone',
    price: 1000,
    description:
      'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Mollitia doloremque architecto quia aliquam? Assumenda a, ipsam quidem enim neque saepe corrupti, fugiat exercitationem vero, optio dolorem alias maxime dolore omnis.'
  },
  {
    id: '2',
    image: '/imgs/products/macbook.jpg',
    name: 'Macbook pro 14 inch 2021 32gb',
    price: 2000,
    description:
      'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Mollitia doloremque architecto quia aliquam? Assumenda a, ipsam quidem enim neque saepe corrupti, fugiat exercitationem vero, optio dolorem alias maxime dolore omnis.'
  },
  {
    id: '3',
    image: '/imgs/products/watch.jpg',
    name: 'Watch',
    price: 500,
    description:
      'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Mollitia doloremque architecto quia aliquam? Assumenda a, ipsam quidem enim neque saepe corrupti, fugiat exercitationem vero, optio dolorem alias maxime dolore omnis.'
  }
]
