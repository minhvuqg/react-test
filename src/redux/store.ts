import { combineReducers } from 'redux'
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'

import productSlice from './reducers/productSlice'

const customizedMiddleware: any = getDefaultMiddleware({
  serializableCheck: false
})

const persistedReducer: any = combineReducers({
  cart: productSlice
})

export default configureStore({
  reducer: persistedReducer,
  middleware: customizedMiddleware
})
