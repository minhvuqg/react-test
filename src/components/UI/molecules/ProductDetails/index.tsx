import { FC } from 'react'

import { convertToCurrency } from 'helpers/string'

import { IProps } from './type.d'

import './style.scss'

const ProductDetails: FC<IProps> = ({ product }: IProps): JSX.Element => {
  return (
    <div className="product-info">
      <div className="image m-r-16">
        <img data-testid="img" src={product.image} alt={product.name} />
      </div>

      <div className="info">
        <div className="name-and-price">
          <span data-testid="name" className="-text-medium">
            {product.name}
          </span>
          <span data-testid="price">{convertToCurrency(product.price)}</span>
        </div>

        <p data-testid="des">{product.description}</p>
      </div>
    </div>
  )
}

export { ProductDetails }
