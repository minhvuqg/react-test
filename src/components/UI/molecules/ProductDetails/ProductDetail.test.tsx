import TestRenderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { ProductDetails } from '../ProductDetails'
import store from 'redux/store'
import { productList } from 'data'

it('render detail correctly', () => {
  const detailEl = TestRenderer.create(
    < Provider store={store} >
      <ProductDetails product={productList[0]} />
    </Provider >).toJSON()
  expect(detailEl).toMatchSnapshot()
})
