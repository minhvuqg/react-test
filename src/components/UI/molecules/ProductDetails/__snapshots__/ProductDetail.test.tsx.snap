// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render detail correctly 1`] = `
<div
  className="product-info"
>
  <div
    className="image m-r-16"
  >
    <img
      alt="Iphone"
      data-testid="img"
      src="/imgs/products/iphone.png"
    />
  </div>
  <div
    className="info"
  >
    <div
      className="name-and-price"
    >
      <span
        className="-text-medium"
        data-testid="name"
      >
        Iphone
      </span>
      <span
        data-testid="price"
      >
        $1,000.00
      </span>
    </div>
    <p
      data-testid="des"
    >
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Mollitia doloremque architecto quia aliquam? Assumenda a, ipsam quidem enim neque saepe corrupti, fugiat exercitationem vero, optio dolorem alias maxime dolore omnis.
    </p>
  </div>
</div>
`;
